### ISSUE TITLE
Update 4.9.34 auf 4.13.10 via Contao Manager: Update Scripts

### ISSUE THREAD

**Issue by berecont** [contao/contao-manager#731]
Contao Manager 1.6.0 updated Scripts, obwohl noch kein Update durchgeführt wird:
```
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    },
```  
wird zu:
```
    "scripts": {
        "post-install-cmd": [
            "@php vendor/bin/contao-setup"
        ],
        "post-update-cmd": [
            "@php vendor/bin/contao-setup"
        ]
    },
```
Update schlägt mit  
 `Script @php vendor/bin/contao-setup --ansi handling the post-update-cmd event returned with error code 1` 
fehl

**Comment by berecont** [comment]
Fehlerausgabe:  
```
composer require contao/manager-bundle=4.13.* contao/news-bundle=4.13.* contao/calendar-bundle=4.13.* contao/faq-bundle=4.13.* contao/comments-bundle=4.13.* contao/newsletter-bundle=4.13.* contao/listing-bundle=4.13.*
$ /usr/local/php74/bin/php -q -dmax_execution_time=0 -dmemory_limit=-1 -dallow_url_fopen=1 -ddisable_functions= -ddate.timezone=Europe/Berlin /var/www/vhosts/mydomain.xy/httpdocs/main/web/contao-manager.phar.php composer require contao/manager-bundle=4.13.* contao/news-bundle=4.13.* contao/calendar-bundle=4.13.* contao/faq-bundle=4.13.* contao/comments-bundle=4.13.* contao/newsletter-bundle=4.13.* contao/listing-bundle=4.13.* --no-update --no-scripts --prefer-stable --sort-packages --no-ansi --no-interaction
/var/www/vhosts/mydomain.xy/httpdocs/main/composer.json has been updated
# Process terminated with exit code 0
# Result: OK

$ /usr/local/php74/bin/php -q -dmax_execution_time=0 -dmemory_limit=-1 -dallow_url_fopen=1 -ddisable_functions= -ddate.timezone=Europe/Berlin /var/www/vhosts/mydomain.xy/httpdocs/main/web/contao-manager.phar.php composer update --with-dependencies --no-install --no-dev --no-progress --no-ansi --no-interaction --optimize-autoloader
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Lock file operations: 17 installs, 54 updates, 9 removals
  - Removing composer/package-versions-deprecated (1.11.99.5)
  - Removing contao-components/highlight (9.18.1)
  - Removing lexik/maintenance-bundle (v2.1.5)
  - Removing michelf/php-markdown (1.9.1)
  - Removing ramsey/uuid (3.9.6)
  - Removing swiftmailer/swiftmailer (v6.3.0)
  - Removing symfony/debug (v4.4.44)
  - Removing symfony/polyfill-iconv (v1.26.0)
  - Removing symfony/swiftmailer-bundle (v3.5.4)
  - Upgrading bugbuster/contao-visitors-bundle (1.6.13 => 1.8.0)
  - Upgrading contao-components/tinymce4 (4.9.12 => 5.10.5)
  - Upgrading contao/calendar-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/comments-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/core-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/faq-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/installation-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/listing-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/manager-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/news-bundle (4.9.34 => 4.13.10)
  - Upgrading contao/newsletter-bundle (4.9.34 => 4.13.10)
  - Locking dflydev/dot-access-data (v3.0.1)
  - Upgrading doctrine/dbal (2.13.9 => 3.4.5)
  - Downgrading doctrine/persistence (3.0.3 => 2.5.4)
  - Locking league/commonmark (2.3.5)
  - Locking league/config (v1.1.1)
  - Locking league/flysystem (2.5.0)
  - Locking league/flysystem-bundle (2.4.0)
  - Locking league/mime-type-detection (1.11.0)
  - Upgrading madeyourday/contao-rocksolid-columns (v2.0.3 => v2.1.1)
  - Upgrading madeyourday/contao-rocksolid-custom-elements (v2.3.12 => v2.4.2)
  - Upgrading madeyourday/contao-rocksolid-icon-picker (v2.0.3 => v2.1.0)
  - Upgrading monolog/monolog (1.27.1 => 2.8.0)
  - Locking nette/schema (v1.2.2)
  - Locking nette/utils (v3.2.8)
  - Locking psr/event-dispatcher (1.0.0)
  - Locking scrivo/highlight.php (v9.18.1.9)
  - Locking spatie/schema-org (3.9.0)
  - Upgrading symfony/asset (v4.4.46 => v5.4.13)
  - Upgrading symfony/cache (v4.4.46 => v5.4.13)
  - Upgrading symfony/config (v4.4.44 => v5.4.11)
  - Upgrading symfony/console (v4.4.45 => v5.4.13)
  - Upgrading symfony/debug-bundle (v4.4.37 => v5.4.11)
  - Upgrading symfony/dependency-injection (v4.4.44 => v5.4.13)
  - Upgrading symfony/doctrine-bridge (v4.4.46 => v5.4.13)
  - Upgrading symfony/dom-crawler (v4.4.45 => v5.4.12)
  - Upgrading symfony/dotenv (v4.4.37 => v5.4.5)
  - Upgrading symfony/error-handler (v4.4.44 => v5.4.11)
  - Upgrading symfony/event-dispatcher (v4.4.44 => v5.4.9)
  - Upgrading symfony/event-dispatcher-contracts (v1.1.13 => v2.5.2)
  - Upgrading symfony/expression-language (v4.4.44 => v5.4.11)
  - Upgrading symfony/finder (v4.4.44 => v5.4.11)
  - Upgrading symfony/framework-bundle (v4.4.46 => v5.4.13)
  - Upgrading symfony/http-client (v4.4.46 => v5.4.13)
  - Upgrading symfony/http-foundation (v4.4.46 => v5.4.13)
  - Upgrading symfony/http-kernel (v4.4.46 => v5.4.13)
  - Locking symfony/intl (v5.4.11)
  - Upgrading symfony/lock (v4.4.46 => v5.4.10)
  - Locking symfony/mailer (v5.4.13)
  - Upgrading symfony/monolog-bridge (v4.4.43 => v5.4.10)
  - Locking symfony/password-hasher (v5.4.11)
  - Locking symfony/polyfill-uuid (v1.26.0)
  - Upgrading symfony/process (v4.4.44 => v5.4.11)
  - Upgrading symfony/proxy-manager-bridge (v4.4.39 => v5.4.6)
  - Upgrading symfony/routing (v4.4.44 => v5.4.11)
  - Upgrading symfony/security-bundle (v4.4.44 => v5.4.11)
  - Upgrading symfony/security-core (v4.4.46 => v5.4.13)
  - Upgrading symfony/security-csrf (v5.2.12 => v5.4.11)
  - Upgrading symfony/security-guard (v4.4.46 => v5.4.13)
  - Upgrading symfony/security-http (v4.4.44 => v5.4.13)
  - Upgrading symfony/stopwatch (v4.4.46 => v5.4.13)
  - Upgrading symfony/translation (v4.4.45 => v5.4.12)
  - Upgrading symfony/twig-bridge (v4.4.45 => v5.4.12)
  - Upgrading symfony/twig-bundle (v4.4.41 => v5.4.8)
  - Locking symfony/uid (v5.4.13)
  - Upgrading symfony/var-dumper (v4.4.46 => v5.4.13)
  - Upgrading symfony/web-link (v5.2.12 => v5.4.3)
  - Upgrading symfony/web-profiler-bundle (v4.4.44 => v5.4.13)
  - Upgrading symfony/yaml (v4.4.45 => v5.4.12)
  - Locking twig/extra-bundle (v3.4.0)
  - Upgrading twig/twig (v2.15.3 => v3.4.3)
Writing lock file
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package patchwork/utf8 is abandoned, you should avoid using it. Use symfony/polyfill-mbstring or symfony/string instead.
Package true/punycode is abandoned, you should avoid using it. No replacement was suggested.
Package webmozart/path-util is abandoned, you should avoid using it. Use symfony/filesystem instead.
Package zendframework/zend-cache is abandoned, you should avoid using it. Use laminas/laminas-cache instead.
Package zendframework/zend-eventmanager is abandoned, you should avoid using it. Use laminas/laminas-eventmanager instead.
Package zendframework/zend-servicemanager is abandoned, you should avoid using it. Use laminas/laminas-servicemanager instead.
Package zendframework/zend-stdlib is abandoned, you should avoid using it. Use laminas/laminas-stdlib instead.
120 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
contao/manager-plugin: Dumping generated plugins file...
contao/manager-plugin: ...done dumping generated plugins file
Could not open input file: vendor/bin/contao-setup
> @php vendor/bin/contao-setup
Script @php vendor/bin/contao-setup handling the post-update-cmd event returned with error code 1
# Process terminated with exit code 1
# Result: General error  
```  
im Ordner `vendor/bin/` befindet sich keine Datei `contao-setup`

**Comment by ausi** [comment]
`Could not open input file: vendor/bin/contao-setup` deutet darauf hin, dass die installation nicht vollständig war.
Existiert die Datei *vendor/bin/contao-setup* auf dem webspace?

**Comment by fritzmg** [comment]
According to the console output the package update was successful. However the `post-update` script could not be executed because `vendor/bin/contao-setup` did not exist (yet) at that point.

**Comment by fritzmg** [comment]
I cannot really reproduce the issue without the Contao Manager though. As soon as Composer executes

```
  - Upgrading contao/manager-bundle (4.9.34 => 4.13.10)
```

the `vendor/bin/contao-setup` file should also be created by composer.

**Comment by fritzmg** [comment]
Actually I am able to reproduce the problem now:

1. `composer create-project contao/managed-edition . 4.9.*`
2. Manually change the `composer.json` to use the the [new `contao-setup` binary](https://github.com/contao/managed-edition/blob/55ce9e14d0049185ea4f95a22f5c19b098b29b86/composer.json#L30-L37).
3. `composer require contao/manager-bundle=4.13.* contao/news-bundle=4.13.* contao/calendar-bundle=4.13.* contao/faq-bundle=4.13.* contao/comments-bundle=4.13.* contao/newsletter-bundle=4.13.* contao/listing-bundle=4.13.* --no-update --no-scripts --prefer-stable --sort-packages --no-ansi --no-interaction -W`
4. `composer update --with-dependencies --no-install --no-dev --no-progress --no-ansi --no-interaction --optimize-autoloader`

This will result in `Could not open input file: vendor\bin\contao-setup`.

i.e. - with these commands (mirroring the commands used by the Contao Manager) the `vendor/bin/contao-setup` file will not be created.

The cause seems to be the `--no-install` option. Without it the `vendor/bin/contao-setup` binary will be created.

**Comment by fritzmg** [comment]
You could add the `--no-scripts` option. What's the intention behind `--no-install` though?

**Comment by aschempp** [comment]
very good idea! Fixed in d684354eded0e66e906483cdda019bbe6c6e15e4

> What's the intention behind --no-install though?

We're updating the composer.lock before installation, so we can enable the maintenance mode in between.
